packages/@react-aria/interactions/test/useFocusVisible.test.js

```diff
@@ -90,6 +91,136 @@ describe('useFocusVisible', function () {
 
     expect(el.textContent).toBe('example');
   });
+
+  describe('Setups global event listeners in a different window', () => {
+    let iframe;
+    let iframeRoot;
+    beforeEach(() => {
+      iframe = document.createElement('iframe');
+      window.document.body.appendChild(iframe);
+      iframeRoot = iframe.contentWindow.document.createElement('div');
+      iframe.contentWindow.document.body.appendChild(iframeRoot);
+    });
+
+    afterEach(() => {
+      iframe.remove();
+    });
+
+    it('sets up focus listener in a different window', async function () {
+      ReactDOMRender(<Example id="iframe-example" />, iframeRoot);
+      await waitFor(() => {
+        expect(document.querySelector('iframe').contentWindow.document.body.querySelector('div[id="iframe-example"]')).toBeTruthy();
+      });
+      const el = document.querySelector('iframe').contentWindow.document.body.querySelector('div[id="iframe-example"]');
+
+      // Focus in iframe before setupFocus should not do anything
+      fireEvent.focus(iframe.contentWindow.document.body);
+      expect(el.textContent).toBe('example');
+
+      // Setup focus in iframe
+      setupFocus(iframeRoot);
+      expect(el.textContent).toBe('example');
+
+      // Focus in iframe after setupFocus
+      fireEvent.focus(iframe.contentWindow.document.body);
+      expect(el.textContent).toBe('example-focusVisible');
+    });
+
+    it('removes event listeners on beforeunload', async function () {
+      ReactDOMRender(<Example id="iframe-example" />, iframeRoot);
+      setupFocus(iframeRoot);
+
+      await waitFor(() => {
+        expect(document.querySelector('iframe').contentWindow.document.body.querySelector('div[id="iframe-example"]')).toBeTruthy();
+      });
+      const el = document.querySelector('iframe').contentWindow.document.body.querySelector('div[id="iframe-example"]');
+      expect(el.textContent).toBe('example-focusVisible');
+
+      fireEvent.mouseDown(el);
+      expect(el.textContent).toBe('example');
+
+      // Focus events after beforeunload no longer work
```

Comment:
if we export the map of setups from useFocusVisible (but not from the barrel index file), then we could verify that we've removed the entry from the map as well

Reply:
I added a test around that but cast the return type of `useFocusVisible` so I don't expose the property on the type to other devs or should I include `hasSetupGlobalListeners` as a property of `type FocusVisibleResult`?

Reply:
o, no need to return it from useFocusVisible, you can just export it at the module level and not re-export it from 'index.ts'

then you can import it into the test file as
`import {hasSetupGlobalListeners} from '../src/useFocusVisible';`

but outside developers wouldn't be able to do `import {hasSetupGlobalListeners} from '@react-aria/interactions'`

Reply:
Done!